core/domain/commit_commands_domain_test.py

```diff
@@ -0,0 +1,278 @@
+
+        Args:
+            name: str. The command name.
+            parameters: dict. Dict containing command parameters.
+        """
+        commit_cmd_domain_object = self.COMMIT_CMD_CLASS(name, parameters)
+        commit_cmd_domain_object.validate()
+
+
+    def validate_with_missing_keys_in_command(self, name, parameters):
+        """Validates a commit command with missing keys in parameters.
+
+        Args:
+            name: str. The command name.
+            parameters: dict. Dict containing command parameters.
+        """
+
+        if not parameters.keys():
+            return
+        parameter_key = sorted(parameters.keys())[0]
+        parameters.pop(parameter_key, None)
+        commit_cmd_domain_object = self.COMMIT_CMD_CLASS(name, parameters)
+        self.assert_validation_error(
+            commit_cmd_domain_object,
+            'The following required keys are missing: %s' % parameter_key)
+
+
+    def validate_with_extra_keys_in_command(self, name, parameters):
+        """Validates a commit command with extra keys in parameters.
+
+        Args:
+            name: str. The command name.
+            parameters: dict. Dict containing command parameters.
+        """
+        parameters['random'] = 'random'
+        commit_cmd_domain_object = self.COMMIT_CMD_CLASS(name, parameters)
+        self.assert_validation_error(
+            commit_cmd_domain_object,
+            'The following extra keys are present: random')
+
+
+class ExplorationCommitCmdUnitTests(BaseCommitCmdUnitTests):
+    """Test the exploration commit cmd domain object."""
+
+    def setUp(self):
+        super(ExplorationCommitCmdUnitTests, self).setUp()
+        self.COMMIT_CMD_CLASS = commit_commands_domain.ExplorationCommitCmd
+        self.COMMAND_LIST = [{
+            'name': exp_domain.CMD_CREATE_NEW,
+            'parameters': {'category': 'category', 'title': 'title'}
+        }, {
+            'name': exp_domain.CMD_ADD_STATE,
+            'parameters': {'state_name': 'state'}
+        }, {
+            'name': exp_domain.CMD_DELETE_STATE,
+            'parameters': {'state_name': 'state'}
+        }, {
+            'name': exp_domain.CMD_RENAME_STATE,
+            'parameters': {
+                'old_state_name': 'old_state', 'new_state_name': 'new_state'
+            }
+        }, {
+            'name': exp_domain.CMD_EDIT_STATE_PROPERTY,
+            'parameters': {
+                'property_name': 'property', 'state_name': 'state',
+                'new_value': 'value'
+            }
+        }, {
+            'name': exp_domain.CMD_EDIT_EXPLORATION_PROPERTY,
+            'parameters': {
+                'property_name': 'property', 'new_value': 'new_value',
+                'old_value': 'old_value'
+            }
+        }, {
+            'name': exp_domain.CMD_MIGRATE_STATES_SCHEMA_TO_LATEST_VERSION,
+            'parameters': {'from_version': 1, 'to_version': 3}
+        }]
+
+    def test_commands(self):
```

Comment:
Use a clearer test name following [these guidelines](https://github.com/oppia/oppia/wiki/Writing-backend-tests#guidelines-for-writing-good-tests). Split into multiple test functions if necessary (perhaps one testing validity, one testing missing keys, etc.)

Btw, in general, this test file has a bit too much logic in it. When writing tests, try to keep the code really simple and straightforward, to minimize the chance of errors in the test code. It's OK if there's some repetition.

Reply:
Updated the tests.